docs/src/proposals/epoch_accounts_hash.md

```diff
@@ -0,0 +1,203 @@
+---
+title: Epoch Accounts Hash
+---
+
+*Paraphrasing from https://github.com/solana-labs/solana/issues/26847*
+
+## Background
+
+Rent collection checks every account at least once per epoch.  At each slot, a
+deterministic set of accounts (based on the pubkey range) is loaded, checked
+for rent collection, and stored back to the Accounts DB at this new slot.
+
+Accounts are stored (rewritten) _even if_ they are unchanged.  This has a few
+positive effects.
+  1. Once an account is rewritten, the previous version at an older slot is now
+     dead. As entire slots and AppendVecs become full of only dead accounts,
+     they can then be dropped/recycled.
+  2. Each account rewritten due to rent collection is included in that slot's
+     bank hash.  Since the bank hash is part of what is voted on for consensus,
+     this means every account is verified by the network at least once per
+     epoch.
+
+However, there is a big downside to rewriting unchanged accounts: performance.
+Storing accounts can be very expensive.  And since accounts now are required to
+be rent-exempt, the majority of accounts rewritten due to rent collection are
+unchanged.  What if unchanged accounts were no longer rewritten? This would
+minimally be a big performance win.
+
+
+## Problem
+
+If rent collection no longer is rewriting unchanged accounts, we lose the two
+positive effects.  Dealing with Positive Effect 1 (from above) will be handled
+by _Ancient AppendVecs_, and will not be discussed here.  So how do we still
+get the security from Positive Effect 2?  How can we still verify every account
+at least once per epoch, *as part of consensus*, but without rewriting
+accounts?
+
+
+## Proposed Solution
+
+Perform a full accounts hash calculation once per epoch, and hash the result
+into a bank's `hash`.  This will be known as the _Epoch Accounts Hash_, or
+_EAH_.
+
+This retains the Positive Effect 2 from rent collection by checking every
+account at least once per epoch.  Thus, any validators with missing, corrupt,
+or extra accounts will identify those issues within 1-2 epochs.
+
+
+### Implementation
+
+Performing a full accounts hash takes a relatively long time to complete.  For
+this reason, the EAH calculation must take place in the background.
+
+In order for all the validators to calculate the same accounts hash, the
+calculation must be based on the same view of all accounts.  This means the EAH
+must be based on a predetermined slot.  This will be known as the `start slot`.
+The `start slot` is calculated as an offset from the first slot of an epoch.
```

Comment:
More precisely: `start slot` is the first root >= epoch first slot (whether a root or not) + slot offset